### PR TITLE
feat: make https repo credentials editable in the UI

### LINKED ISSUE #9108: Make Repos and other setting items editable in the Argo CD UI
# Summary

We got feedback from an Argo CD user that it would be nice to have as many as the settings to be editable.

# Motivation

Currently it's possible to add/delete things in the repo list, but not update.

Clusters and Projects are editable in UI now, I would like to be able to update repos and other settings using the Argo CD UI

(https://cd.apps.argoproj.io/settings/repos) 

# Proposal

Create a repo details page which allows updating selected repo information.

### PR DESCRIPTION
**Pull request by ciiay**
signed-off: @ciiay 
fixes: #9108 

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)). 

Screenshot:
Screen Shot 2022-06-24 at 9 26 38 AM
Screen Shot 2022-06-24 at 9 27 25 AM

### CODE REVIEW

**Comment by reginapizza** [comment]
Hi Yi! Just tried this out and it works great but I have one little nit. When the validation error pops up, usually they are on the line below the input. Here it is in the text input itself, making you not able to click on it to get the cursor (see video below). I would suggest making the validation errors like the other text inputs, as seen in the second picture here. Otherwise looks great so far, I'll continue to look into it code-wise! :smile: 

https://user-images.githubusercontent.com/50851526/176242712-cfb5733a-2882-4514-8a58-f7645278418c.mp4

ex 1

**Comment by ciiay** [comment]
Hi @reginapizza , thanks for the review. Good catch on that unclickable input field. I actually reused a current `<EditablePanel>` component and its style. So it's an existing issue for wherever this component is used. I fixed it at my new PR update, it should fix the issue for where else it's used too.

e.g. Not clickable input field before my fix here too
Screen Shot 2022-06-28 at 4 11 58 PM

Regarding the location of this error message, if I make it underneath the input field, it will stretch out the entire section which doesn't look good, so I prefer to keep it inside the input as it's an existing style already. How do you think?

*no error message: small and neat section
Screen Shot 2022-06-28 at 4 25 15 PM
v.s. with the error message: stretched out section with extra thick bottom margin
Screen Shot 2022-06-28 at 4 25 07 PM

File: ui/src/app/shared/components/editable-panel/editable-panel.scss [hunk 1]
```diff
@@ -3,6 +3,10 @@
 .editable-panel {
     position: relative;
 
+    input {
+        z-index: 10;
+    }
+
```
Comment (ciiay) [inline ui/src/app/shared/components/editable-panel/editable-panel.scss:9]: @reginapizza Here's the fix for unclickable input field issue.

**Review by saumeya — COMMENTED** [review]
Great PR Yi, left a small comment

File: ui/src/app/settings/components/repo-details/repo-details.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        enableLfs: repo.enableLfs || false,
+        proxy: repo.proxy || '',
+        project: repo.project || ''
+    };
+
+    return (
+        <EditablePanel
+            values={repo}
+            validate={input => ({
+                password: !input.password && input.username && 'Password is required if username is given.'
```
[the remark this answers is not in the document]
Comment (ciiay) [reply ui/src/app/settings/components/repo-details/repo-details.tsx:74]: Thanks for the review. I see the same thing when I wrote it but I aligned up with the current validation for the creation sliding panel. It only checks one way, so I kept it consistent. Do you think I should update both places?
Screen Shot 2022-06-30 at 1 23 30 PM

[the remark this answers is not in the document]
Comment (saumeya) [reply ui/src/app/settings/components/repo-details/repo-details.tsx:74]: Maybe better to update both places for better user experience?

**Review by saumeya — APPROVED** [review]
LGTM

**Review by rbreeze — APPROVED** [review]
LGTM

**Pull request merged**